src/officialValidators/week.js

```diff
@@ -3,20 +3,41 @@
 import { parseWeek } from '../utils';
 
 // Example week: "2016-W33"
-// TODO: this needs to be improved since some years have 52 weeks
-const MAX_WEEK = 53
 export default function weekValidator(control) {
   // Special case for empty values. This is the job of the `required` validator
   if (!control.value) {
     return false;
   }
 
-  const value = control.value
-  const [year, week] = parseWeek(value)
+  const value = control.value;
+  const [year, week] = parseWeek(value);
 
   if (!year || !week) {
-    return true
+    return true;
   }
 
-  return !(0 < year && 1 <= week && week <= MAX_WEEK)
+  return !(0 < year && 1 <= week && week <= weeksInYear(year));
+}
+
+function weeksInYear(year) {
+  var d = new Date(year, 11, 31);
```

Comment:
I would go to something like `let date` in here. Or even better `let lastDayOfTheYear`.